central/signatureintegration/datastore/datastore_impl.go

```diff
@@ -149,6 +163,67 @@ func (d *datastoreImpl) verifyIntegrationIDDoesNotExist(ctx context.Context, id
 	return nil
 }
 
+func (d *datastoreImpl) verifyIntegrationIDIsNotInPolicy(ctx context.Context, id string) error {
+	var useAllAccessCtx bool
+	if err := sac.VerifyAuthzOK(policySAC.ReadAllowed(ctx)); err != nil {
+		useAllAccessCtx = true
+	}
+
+	// Only create a context with all access if we are not allowed to read policies.
+	policyCtx := ctx
+	if useAllAccessCtx {
```

Comment:
I think what we probably want to do instead is something like this unconditionally:
```
	policyCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
		sac.AllowFixedScopes(sac.AccessModeScopeKeys(storage.Access_READ_WRITE_ACCESS),
			sac.ResourceScopeKeys(resources.Policy)))
```

Rationale: We want to do as little hand rolling as possible. For example, if `policy` becomes a scoped resource, this code should ideally keep working.